src/OrchardCore/OrchardCore.DisplayManagement/Shapes/DateTimeShapes.cs

```diff
@@ -122,58 +131,76 @@ public IHtmlContent Duration(TimeSpan? timeSpan)
             return HtmlString.Empty;
         }
 
-        var days = (int)timeSpan.Value.TotalDays;
-        var hours = timeSpan.Value.Hours;
-        var minutes = timeSpan.Value.Minutes;
-        var seconds = timeSpan.Value.Seconds;
+        var tag = new TagBuilder("span");
+
+        tag.AddCssClass("timespan-preview-value");
+
+        tag.Attributes["title"] = timeSpan.ToString();
 
-        var builder = new HtmlContentBuilder();
+        tag.InnerHtml.AppendHtml(GetDuration(timeSpan.Value));
+
+        return tag;
+    }
+
+    private LocalizedHtmlString GetDuration(TimeSpan timeSpan)
+    {
+        var days = timeSpan.Days;
+        var hours = timeSpan.Hours;
+        var minutes = timeSpan.Minutes;
+        var seconds = timeSpan.Seconds;
 
         if (days > 0)
         {
-            builder.AppendHtml(H.Plural(days, "{1} day", "{1} days", days));
+            return GetDurationInDays(timeSpan, days);
         }
 
         if (hours > 0)
         {
-            if (builder.Count > 0)
-            {
-                builder.AppendWhitespace();
-            }
-            builder.AppendHtml(H.Plural(hours, "{1} hour", "{1} hours", hours));
+            return GetDurationInHours(timeSpan, hours);
         }
 
         if (minutes > 0)
         {
-            if (builder.Count > 0)
-            {
-                builder.AppendWhitespace();
-            }
-
-            builder.AppendHtml(H.Plural(hours, "{1} minute", "{1} minutes", minutes));
+            return GetDurationInMinutes(timeSpan, minutes);
         }
 
-        if (seconds > 0)
+        return H.Plural(seconds, "1 second", "{0} seconds");
```

Comment:
Handle the final case, when it's less than a second: `"less than a second"`. Otherwise it will show `0 seconds` in some languages.

Reply:
done